core/src/test/scala/unit/kafka/server/ReplicationQuotasTest.scala

```diff
@@ -0,0 +1,187 @@
+
+    //Add data equally to each partition
+    producer = TestUtils.createNewProducer(TestUtils.getBrokerListStrFromServers(brokers), retries = 5, acks = 0)
+    (0 until msgCount).foreach { x =>
+      (0 to 7).foreach { partition =>
+        producer.send(new ProducerRecord(topic, partition, null, msg)).get
+      }
+    }
+
+    //Ensure data is fully written: broker 1 has partition 1, broker 2 has partition 2 etc
+    (0 to 5).foreach { id => waitForOffsetsToMatch(msgCount, id, 100 + id) }
+    //Check the non-throttled partitions too
+    waitForOffsetsToMatch(msgCount, 6, 100)
+    waitForOffsetsToMatch(msgCount, 7, 101)
+
+    val start = System.currentTimeMillis()
+
+    //When we create the 2 new, empty brokers
+    brokers = brokers :+ TestUtils.createServer(fromProps(createBrokerConfig(106, zkConnect)))
+    brokers = brokers :+ TestUtils.createServer(fromProps(createBrokerConfig(107, zkConnect)))
+
+    //Wait for non-throttled partitions to replicate first
+    (6 to 7).foreach { id => waitForOffsetsToMatch(msgCount, id, 100 + id) }
+    val unthrottledTook = System.currentTimeMillis() - start
+
+    //Wait for replicas 0,1,2,3,4,5 to fully replicated to broker 106,107
+    (0 to 2).foreach { id => waitForOffsetsToMatch(msgCount, id, 106) }
+    (3 to 5).foreach { id => waitForOffsetsToMatch(msgCount, id, 107) }
+
+    val throttledTook = System.currentTimeMillis() - start
+
+    //Check the recorded throttled rate is what we expect
+    if (leaderThrottle) {
+      (100 to 105).map(brokerFor(_)).foreach { broker =>
+        val metricName = broker.metrics.metricName("byte-rate", LeaderReplication, "Tracking byte-rate for" + LeaderReplication)
+        val measuredRate = broker.metrics.metrics.asScala(metricName).value()
+        info(s"Broker:${broker.config.brokerId} Expected:$throttle, Recorded Rate was:$measuredRate")
+        assertEquals(throttle, measuredRate, fifteenPercentError(throttle))
+      }
+    } else {
+      (106 to 107).map(brokerFor(_)).foreach { broker =>
+        val metricName = broker.metrics.metricName("byte-rate", FollowerReplication, "Tracking byte-rate for" + FollowerReplication)
+        val measuredRate = broker.metrics.metrics.asScala(metricName).value()
+        info(s"Broker:${broker.config.brokerId} Expected:$throttle, Recorded Rate was:$measuredRate")
+        assertEquals(throttle, measuredRate, fifteenPercentError(throttle))
+      }
+    }
+
+    //Check the times for throttled/unthrottled are each side of what we expect
+    info(s"Unthrottled took: $unthrottledTook, Throttled took: $throttledTook")
+    assertTrue(s"Unthrottled replication of ${unthrottledTook}ms should be < ${expectedDuration * 1000}ms",
+      unthrottledTook < expectedDuration * 1000)
+    assertTrue((s"Unthrottled replication of ${throttledTook}ms should be < ${expectedDuration * 1000}ms"),
```

Comment:
This message doesn't match the test. Shouldn't it be "Throttled Replication of N ms should > M ms"